src/main/java/eu/darkbot/api/game/entities/FakeEntity.java

```diff
@@ -0,0 +1,51 @@
+package eu.darkbot.api.game.entities;
+
+//TODO: Move to API repo
+/**
+ * Fake entities helps to keep virtual entities on map that placed out of visibility range.
+ * It should use when need to assist or predict position these entities for active module.
+ * Cases of use: keeping mine locations for choose better path and prevent release it,
+ *               keeping box locations that should be collected after more prefer tasks.
+ */
+public interface FakeEntity extends Entity {
+    static boolean isFakeEntity(Entity entity) {
+        if (entity instanceof FakeEntity.FakeBox) return true;
+        if (entity instanceof FakeEntity.FakeMine) return true;
+        return false;
+    }
```

Comment:
```suggestion
    static boolean isFakeEntity(Entity entity) {
        return entity instanceof FakeEntity;
    }
```
If the Fake* classes actually inherit the interface, it can be simplified.

Could also actually include generic fake-stuff behavior in here,
i see a good candidate being `setEnabled(true/false)` where if you disable the fake entity it behaves as if it wasn't there at all (ie: starts returning false in isValid()) and you can just arbitrarily set it back to true later to make it relevant again. setTimeout and setRemoveDistance also seem good candidates for the generic FakeEntity

Reply:
implemented in API PR

Reply:
I not want allow hiding entities from user code. I mean closed lifecycle where fake entity is required - it created and add on map immediately and when it nor required then it removing without possibility add it again. Only creating new instance.